src/menu/wallet_menu.c

```diff
@@ -126,15 +127,15 @@ static void ignore_p0_handler(engine_ctx_t *ctx,
 /*****************************************************************************
  * STATIC VARIABLES
  *****************************************************************************/
-static uint8_t wallet_index = 0xFF;
+static uint8_t wallet_id[WALLET_ID_SIZE] = {0};
 
 static const flow_step_t wallet_step = {.step_init_cb = wallet_menu_initialize,
                                         .p0_cb = ignore_p0_handler,
                                         .ui_cb = wallet_menu_handler,
                                         .usb_cb = NULL,
                                         .nfc_cb = NULL,
                                         .evt_cfg_ptr = &device_nav_evt_config,
-                                        .flow_data_ptr = &wallet_index};
+                                        .flow_data_ptr = &wallet_id};
```

Comment:
Is this required?

Reply:
I was not using the flow_data_ptr earlier, added usage to initialize and handler functions. 
I think it's better to keep as many APIs as possible independent of global variables.